sys/net/gnrc/sock/ip/gnrc_sock_ip.c

```diff
@@ -90,20 +90,21 @@ ssize_t sock_ip_recv(sock_ip_t *sock, void *data, size_t max_len,
                      uint32_t timeout, sock_ip_ep_t *remote)
 {
     void *pkt = NULL, *ctx = NULL;
-    ssize_t res;
+    uint8_t *ptr = data;
+    ssize_t res, ret = 0;
+    bool nobufs = false;
 
     assert((sock != NULL) && (data != NULL) && (max_len > 0));
-    res = sock_ip_recv_buf(sock, &pkt, &ctx, timeout, remote);
-    if (res >= 0) {
+    while ((res = sock_ip_recv_buf(sock, &pkt, &ctx, timeout, remote)) > 0) {
         if (res > (ssize_t)max_len) {
-            res = -ENOBUFS;
+            nobufs = true;
+            continue;
```

Comment:
`break`?
Otherwise this will loop forever.

Reply:
Ah never mind - the next iteration will free `ctx`

Reply:
No, we want to have another call to `sock_ip_recv_buf` to free the buffer space ;-).